lib/endpoints/class-wp-rest-terms-controller.php

```diff
@@ -91,6 +91,20 @@ public function get_items( $request ) {
 			}
 		}
 
+		/**
+		 * Filter the query arguments, before passing them to `get_terms()`.
+		 *
+		 * Enables adding extra arguments or setting defaults for a terms
+		 * collection request.
+		 *
+		 * @see https://developer.wordpress.org/reference/functions/get_terms/
+		 *
+		 * @param array           $prepared_args Array of arguments to be
+		 *                                       passed to get_terms.
+		 * @param WP_REST_Request $request       The current request.
+		 */
+		$prepared_args = apply_filters( 'rest_terms_query', $prepared_args, $request );
```

Comment:
This filter naming matches other similar filters, but `pre_rest_terms_query` would be a better match for Core's naming convention.  @rmccue @joehoyle thoughts?  Should we change these now for Posts, Users, etc?